### PR TITLE
[StripeUICore][Identity] Add a disabled state for OTP text fields

### PR DESCRIPTION
**Pull request by ccen-stripe**
## Summary

Add a disabledBackgroundColor and apply to `OneTimeCodeTextField`. 

## Motivation

There's no disabled state for OTP, add the color from [InputFormColors.disabledBackgroundColor](https://github.com/stripe/stripe-ios/blob/1b09a455d04b2fa889fe3f8c8ed680bdbdc15a3d/StripeUICore/StripeUICore/Source/Helpers/InputFormColors.swift#L39)

## Testing

- [x] Unit test
- [x] Manually verified

## Recording
# Screenshots
| Before  | After |
| ------------- | ------------- |
| disableBefore | disableAfter |
| disableBefore | disableAfter |

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/NetworkingOTPView/NetworkingOTPView.swift [hunk 1]
```diff
@@ -129,7 +129,6 @@ final class NetworkingOTPView: UIView {
 
     private func userDidEnterValidOTPCode(_ otpCode: String) {
         otpTextField.resignFirstResponder()
-        // TODO(kgaidis): consider implementing a loading/grayed-out state for `otpTextField`
```
Comment (ccen-stripe) [inline StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/NetworkingOTPView/NetworkingOTPView.swift:132]: @kgaidis-stripe  - Removed this as it's implemented, lmk if I should keep it :)

**Comment by ccen-stripe** [comment]
@stripe/stripe-identity-observers

File: StripeIdentity/StripeIdentity/Source/NativeComponents/Views/PhoneOtpView.swift [hunk 2]
```diff
@@ -59,7 +59,7 @@ class PhoneOtpView: UIView {
         return stackView
     }()
 
-    var viewModel: ViewModel = .InputtingOTP
+    var viewModel: ViewModel?
```
[the remark this answers is not in the document]
Comment (ccen-stripe) [reply StripeIdentity/StripeIdentity/Source/NativeComponents/Views/PhoneOtpView.swift:62]: When just initialized it should not really be `InputtingOTP`, the [inputting field](https://github.com/stripe/stripe-ios/blob/b338d040/StripeIdentity/StripeIdentity/Source/NativeComponents/Views/PhoneOtpView.swift#L109) should be disabled. We could either introduce a new state or a nil state and I just chose the later:)

File: StripeIdentity/StripeIdentity/Source/NativeComponents/Views/PhoneOtpView.swift [hunk 3]
```diff
@@ -112,12 +112,12 @@ class PhoneOtpView: UIView {
...
             submittingIndicator.stopAnimating()
         case .SubmittingOTP:
-            otpTextField.isEnabled = true
+            otpTextField.isEnabled = false
             otpErrorField.isHidden = true
             submittingIndicator.isHidden = false
             submittingIndicator.startAnimating()
         case .ErrorOTP:
-            otpTextField.value = ""
+            otpTextField.performInvalidCodeAnimation()
```
Comment (kgaidis-stripe) [inline StripeIdentity/StripeIdentity/Source/NativeComponents/Views/PhoneOtpView.swift:120]: I guess this is a new change in behavior where an invalid input will preserve the OTP code? Is this coming from design feedback?

File: StripeIdentity/StripeIdentity/Source/NativeComponents/Views/PhoneOtpView.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (kgaidis-stripe) [reply StripeIdentity/StripeIdentity/Source/NativeComponents/Views/PhoneOtpView.swift:62]: oh wait, this is `StripeIdentity` (not Stripe Core component), sorry about that  so never mind

File: StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/NetworkingOTPView/NetworkingOTPView.swift (hunk 1 above)
Comment (kgaidis-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/NetworkingOTPView/NetworkingOTPView.swift:132]: if you will make changes, feel free to keep this, but its not a big deal

the core thing I didn't understand is that this was mostly an Identity change, so Financial Connections actually is unaffected at this current moment without more code changes

File: StripeUICore/StripeUICore/Source/Elements/ElementsUI.swift [hunk 4]
```diff
@@ -95,6 +102,7 @@ import UIKit
         public var primary = UIColor.systemBlue
         public var parentBackground = UIColor.systemBackground
         public var background = ElementsUI.backgroundColor
+        public var disabledBackground = ElementsUI.disabledBackgroundColor
```
Comment (kgaidis-stripe) [inline StripeUICore/StripeUICore/Source/Elements/ElementsUI.swift:105]: might want to check with Elements folks on whether there are any standards around which colors/options we give - am not sure; ex. @yuki-stripe

File: StripeIdentity/StripeIdentity/Source/NativeComponents/Views/PhoneOtpView.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (ccen-stripe) [reply StripeIdentity/StripeIdentity/Source/NativeComponents/Views/PhoneOtpView.swift:62]: Yeah it's Identity specific. When initialized, we first request OTP through `generateOtp`, the state will transition to `RequestingOTP`. Until OTP is generated we shouldn't transition to `InputtingOTP`

**Review by kgaidis-stripe — APPROVED** [review]
looks good; apologies for my confusion on Identity vs. Stripe Core component. might want to check with elements on the new color definition

File: StripeUICore/StripeUICore/Source/Elements/ElementsUI.swift (hunk 4 above)
Comment (ccen-stripe) [reply StripeUICore/StripeUICore/Source/Elements/ElementsUI.swift:105]: This is value is copied from [InputFormColors.disabledBackgroundColor](https://github.com/stripe/stripe-ios/blob/1b09a455d04b2fa889fe3f8c8ed680bdbdc15a3d/StripeUICore/StripeUICore/Source/Helpers/InputFormColors.swift#L39 )

**Pull request merged**